MINOR: Add property to configure showing of standard streams in Gradle

ijuma opened the pull request:
This is handy when debugging certain kinds of Jenkins failures.

ijuma committed:
Fix indenting in build.gradle

ijuma committed:
Add property to configure showing of standard streams in Gradle

ijuma commented:
Review by @ewencp

ewencp commented:
@ijuma LGTM. Should we start a section in the README with available options? When we added maxParallelForks we continued the trend of adding examples, but as we add more simple options like this it might make sense to just have a list in the README.

ijuma commented:
Good point, will do.

ijuma committed:
Document common build options

As suggested by Ewen.

ijuma commented:
@ewencp I introduced a list of options to the readme as you suggested. Let me know what you think.

ijuma committed:
Replace "Publishing to a local repository" with "Install the jars to the local Maven repo"

The latter is is more common and the build options explain how to do the former for
those who need it.

ijuma committed:
Minor grammar fix

guozhangwang commented on README.md:
```diff
@@ -103,8 +103,8 @@ Please note for this to work you should create/update `~/.gradle/gradle.properti
     signing.password=
     signing.secretKeyRingFile=
 
-### Publishing the jars without signing to a local repository ###
```
Originally @ewencp added this for helping other developers work around signing. I will leave it to him for reviewing this.

guozhangwang commented:
LGTM. Leave @ewencp to determine whether we remove the local publishing command.

ewencp commented on README.md:
```diff
@@ -103,8 +103,8 @@ Please note for this to work you should create/update `~/.gradle/gradle.properti
 
-### Publishing the jars without signing to a local repository ###
-    ./gradlew -Dorg.gradle.project.skipSigning=true -Dorg.gradle.project.mavenUrl=file://path/to/repo uploadArchivesAll
+### Installing the jars to the local Maven repository ###
```
If I recall correctly, one of the reasons I included this example with `uploadArchivesAll` was because `./gradlew install` doesn't generate all the artifacts. So if you're trying to make sure everything works and you'll be able to do a release but don't want to do a real upload of the resulting archives, you need to do what I previously had here.

It's possible nobody else cared much about this and I did since I did Confluent's packages for Kafka and ran into this then, but I've also used this as part of my sanity check on build script changes to make sure everything still works. Not sure whether it's important enough toe keep given the documentation of options this patch is adding.

ijuma replied:
Yeah, I should use `installAll` instead, good point. The reason for removing this (as discussed offline) is that it's easy to use it instead of `installAll` by mistake and explaining the difference seemed like it was not worth it.

ijuma committed:
Use `installAll` instead of `install`

ijuma commented:
Changed `install` to `installAll`

ewencp commented:
Great, LGTM too. Thanks @ijuma!
